rcp/org.testeditor.rcp4.views.tcltestrun/src/org/testeditor/rcp4/views/tcltestrun/ReRunLastTestLaunchHandler.xtend

```diff
@@ -0,0 +1,34 @@
+/*******************************************************************************
+ * Copyright (c) 2012 - 2016 Signal Iduna Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ * 
+ * Contributors:
+ * Signal Iduna Corporation - initial API and implementation
+ * akquinet AG
+ * itemis AG
+ *******************************************************************************/
+package org.testeditor.rcp4.views.tcltestrun
+
+import org.eclipse.e4.core.contexts.IEclipseContext
+import org.eclipse.e4.core.di.annotations.CanExecute
+import org.eclipse.e4.core.di.annotations.Execute
+
+class ReRunLastTestLaunchHandler {
+
+	@Execute
+	def reLaunchLastTestLaunch(IEclipseContext context) {
+		val tclLauncherUi = context.get(TclLauncherUi)
+		val lastTestLaunch = context.get(LastTestLaunchInformation)
+		tclLauncherUi.launchTest(lastTestLaunch.testCasesCommaList, lastTestLaunch.project, lastTestLaunch.launcher,
```

Comment:
why not use TestLaunchInformation as parameter (without accessing each field individually) => tclLauncherUi.launchTest(lastTestLanch)